vaadin-upload-flow-parent/vaadin-upload-testbench/src/main/java/com/vaadin/flow/component/upload/testbench/UploadElement.java

```diff
@@ -147,17 +145,10 @@ public void removeFile(int index) {
      *            the number of seconds to wait for the upload to finish
      */
     private void waitForUploads(int maxSeconds) {
-        Timeouts timeouts = getDriver().manage().timeouts();
-        timeouts.setScriptTimeout(maxSeconds, TimeUnit.SECONDS);
-
-        String script = "var callback = arguments[arguments.length - 1];"
-                + "var upload = arguments[0];"
-                + "window.setTimeout(function() {"
-                + "  var inProgress = upload.files.filter(function(file) { return file.uploading;}).length >0;"
-                + "  if (!inProgress) callback();" //
-                + "}, 500);";
-        getCommandExecutor().getDriver().executeAsyncScript(script, this);
+        String script = "return arguments[0].files.every((file) => !file.uploading);";
 
+        waitUntil(driver -> (Boolean) getCommandExecutor().getDriver()
+                .executeScript(script, UploadElement.this), maxSeconds);
```

Comment:
```suggestion
        waitUntil(driver -> (Boolean) getCommandExecutor()
                .executeScript(script, UploadElement.this), maxSeconds);
```

Reply:
`driver.executeScript` does not work because `driver` is a `WebDriver` instead of `TestBenchDriverProxy`.
I'll go for the first suggestion

Reply:
done